dkan_harvest.module

```diff
@@ -1,236 +1,677 @@
 <?php
-
 /**
  * @file
- * Harvesting module for dkan.
+ * Code for the DKAN Harvest feature.
  */
 
+include_once 'dkan_harvest.features.inc';
+
+// TODO Probably move this to drupal variables.
 define(
   'DKAN_HARVEST_CACHE_DIR',
   'public://dkan-harvest-cache'
 );
 
+define(
+  'DKAN_HARVEST_MIGRATION_PREFIX',
+  'dkan_harvest_migrate_'
+);
+
 /**
- * Returns configuration for harvest sources.
+ *
  */
-function dkan_harvest_sources_definition() {
-  $modules = module_implements('harvest_sources');
-  $subscribed_values = array();
-  // Get a list of licenses to display.
-  foreach ($modules as $module) {
-    $function = $module . '_harvest_sources';
-    $values = $function();
-    foreach ($values as $key => $value) {
-      // Make sure other module didn't provide the same source.
-      if (!isset($subscribed_values[$key])) {
-        $subscribed_values[$key] = $value;
-      }
-    }
+function dkan_harvest_menu() {
+  $items = array();
+
+  $items['node/%node/events'] = array(
+    'title' => 'Event Log',
+    'page callback' => 'dkan_harvest_page_event_log',
+    'page arguments' => array(1),
+    'access callback' => 'dkan_harvest_page_event_log_access',
+    'access arguments' => array(1),
+    'type' => MENU_LOCAL_TASK,
+  );
+
+  return $items;
+}
+
+/**
+ * Determines whether the current user may access the harvest source event log
+ * page.
+ *
+ * @param $node The harvest source node being acted upon.
+ *
+ * @return TRUE if access is granted. Otherwise FALSE.
+ */
+function dkan_harvest_page_event_log_access($node) {
+  // Only available for harves source type nodes.
+  if ($node->type != 'harvest_source') {
+    return FALSE;
+  }
+
+  // If the user have access to the harvest node. Grant her/him access to the
+  // event log.
+  return node_access('view', $node);
+}
+
+/**
+ * Get the last event log from a harvest source node.
+ *
+ * @param $node The harvest source node being acted upon.
+ *
+ * @return a date.
+ */
+function dkan_harvest_get_last_event_log($node) {
+  $harvestSource = HarvestSource::getHarvestSourceFromNode($node);
```

Comment:
When we refactor the HarverstSource::_contruct() method to just take a machine name, this will not be necesarry; we can just call `$harvestSource = new HarvestSource($node->field_machine_name['und'][0]['value']);` or whatever.